tokio/src/io/driver/registration.rs

```diff
@@ -232,7 +239,7 @@ cfg_io_readiness! {
             pin!(fut);
 
             crate::future::poll_fn(|cx| {
-                if self.handle.inner().is_none() {
+                if self.handle.inner().is_none() || self.handle.is_shutdown() {
```

Comment:
This seems a little unfortunate since `self.handle` also calls `self.handle.inner()`, so we end up upgrading inner twice.

Reply:
Ah yeah good catch. Should be fixed in 33fd64ebe08ee.